src/tail/log_server.rs

```diff
@@ -0,0 +1,62 @@
+use hyper::server::conn::AddrIncoming;
+use hyper::server::Builder;
+use hyper::service::{make_service_fn, service_fn};
+use hyper::{Body, Method, Request, Response, Server, StatusCode};
+use tokio::sync::oneshot::Receiver;
+
+pub struct LogServer {
+    server: Builder<AddrIncoming>,
+    rx: Receiver<()>,
+}
+
+/// LogServer is just a basic HTTP server running locally; it listens for POST requests on the root
+/// path and simply prints the JSON body of each request as its own line to STDOUT.
```

Comment:
there's nothing preventing some other process from making requests to this server or the tunnel; this is unlikely to become an issue but may be worth calling out.

Reply:
I don't think there would be. Let's assume a malicious actor has root access to a victims machine and deploys a binary that sends requests to this server, what can they do? All the server does is print out any POST requests to stdout, so I guess they could spam your terminal but otherwise it doesn't look like they could pivot to other functionality.

Reply:
guess if they had root access you've got bigger problems, thanks David :)